src/Analyzers/CSharp/Tests/RemoveUnnecessaryCast/RemoveUnnecessaryCastTests.cs

```diff
@@ -7319,99 +7827,138 @@ void M(bool x)
     }
 }";
 
-            await VerifyCS.VerifyCodeFixAsync(source, source);
+            await new VerifyCS.Test
+            {
+                TestCode = source,
+                FixedCode = source,
+                LanguageVersion = LanguageVersion.CSharp8
+            }.RunAsync();
         }
 
         [WorkItem(20211, "https://github.com/dotnet/roslyn/issues/21613")]
         [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryCast)]
-        public async Task DoNotRemoveNecessaryCastInConditional5()
+        public async Task DoRemoveUnnecessaryCastInConditional4()
```

Comment:
a lot of cases are no supported in C# 9 thanks to target typing.  the tests were duplicated and split to test the C#8 and C#9 behavior.